docs/clustering.rst

```diff
@@ -0,0 +1,224 @@
+Clustering
+===========
+
+Merlin allows clustering of multiple Naemon instances, to add redundancy and
+loadbalancing to the Naemon monitoring infrastructure.
+
+Merlin works in two main ways:
+- Sends Naemon events, such as check results, between nodes over a TCP connection
+- Keeps track, and syncs Naemon configuration across nodes over SSH
+
+A single Merlin node, can have up to 65534 neighbours, allowing one to build
+a large monitoring infrastructure.
+
+
+Node types
+-----------
+
+Merlin has three different node types:
+
+Peer
+^^^^^
+
+A peer is a redundant and loadbalanced node. A collection of peers, called a
+peergroup, divides the check load equally. If you have three nodes in a peergroup,
+each peer will execute 33% of the configured Naemon checks each. If one of the
+nodes in a peergroup goes down, Merlin will redistribute the check load to the
+two online peers, so they execute 50% of the checks each.
+
+Peers will always run the same Naemon configuration.
+
+Poller
+^^^^^^^
+
+A poller is responsible for checking host and services belonging to one or more
+hostgroups. This makes it possible to put a Merlin node in a remote network. Each
+poller can also be peered, ensuring multiple nodes are responsible for the set
+of hostgroups.
+
+A poller only has a subset of the full Naemon configuration (based on the
+hostgroup assignment), and will never know of any objects not belonging to the
+specified hostgroups.
+
+It is possible to setup multiple pollergroups which are responsible for different
+sets of hostgroups.
+
+Master
+^^^^^^^
+
+A poller must connect to each node in the master peergroup. The master
+peergroup, keeps track of the full Naemon configuration set. Masters will split
+the Naemon configuration for pollers to use.
+
+Cluster setup walkthrough
+--------------------------
+
+In this walk-through we'll go through creating a cluster with three nodes in
+total. A master peergroup with two masters, and a single remote poller.
+
+By the end we should have a node structure that looks like the following:
+
+.. code-block:: none
+
+    +----------+    +----------+
+    | master01 |----| master02 |
+    +----------+    +----------+
+       |
+       |
+       | HOSTGROUP: pollergroup  +----------+
+       = ------------------------| poller01 |
+                                 +----------+
+
+Preparation
+^^^^^^^^^^^^
+
+To begin with, prepare three machines (master01, master02, poller01), with
+Naemon and Merlin installed as per the installation instructions (LINK). Make
+sure that there are no firewalls blocking port 22 (for SSH) & 15551 (Merlins
+default TCP port). Ensure that you had the root password for all machines at
+hand.
```

Comment:
I guess also that root login, using password, is required between peers to continue setup as documented. That is not always the case on systems (for good reasons too).

Reply:
as in: clarify that SSH root login with password should be allowed?